src/Resources/contao/classes/Frontend.php

```diff
@@ -654,224 +654,16 @@ public static function indexPageIfApplicable(Response $objResponse)
-		if ($expire < time())
-		{
-			ob_end_clean();
-
-			return null;
-		}
-
-		// Define the static URL constants (see #7914)
-		define('TL_FILES_URL', $files);
-		define('TL_ASSETS_URL', $assets);
-
-		// Read the buffer
-		$strBuffer = ob_get_clean();
-
-		/** @var AttributeBagInterface $session */
-		$session = \System::getContainer()->get('session')->getBag('contao_frontend');
-
-		// Session required to determine the referer
-		$data = $session->all();
-
-		// Set the new referer
-		if (!isset($_GET['pdf']) && !isset($_GET['file']) && !isset($_GET['id']) && $data['referer']['current'] != \Environment::get('requestUri'))
-		{
-			$data['referer']['last'] = $data['referer']['current'];
-			$data['referer']['current'] = substr(\Environment::get('requestUri'), strlen(\Environment::get('path')) + 1);
-		}
-
-		// Store the session data
-		$session->replace($data);
-
-		// Load the default language file (see #2644)
-		\System::loadLanguageFile('default');
-
-		// Replace the insert tags and then re-replace the request_token tag in case a form element has been loaded via insert tag
-		$strBuffer = \Controller::replaceInsertTags($strBuffer, false);
-		$strBuffer = str_replace(array('{{request_token}}', '[{]', '[}]'), array(REQUEST_TOKEN, '{{', '}}'), $strBuffer);
-
-		// HOOK: allow to modify the compiled markup (see #4291 and #7457)
-		if (isset($GLOBALS['TL_HOOKS']['modifyFrontendPage']) && is_array($GLOBALS['TL_HOOKS']['modifyFrontendPage']))
-		{
-			foreach ($GLOBALS['TL_HOOKS']['modifyFrontendPage'] as $callback)
-			{
-				$strBuffer = \System::importStatic($callback[0])->{$callback[1]}($strBuffer, null);
-			}
-		}
-
-		// Content type
-		if (!$content)
-		{
-			$content = 'text/html';
-		}
-
-		$response = new Response($strBuffer);
-
-		// Send the status header (see #6585)
-		if ($type == 'error_403')
-		{
-			$response->setStatusCode(Response::HTTP_FORBIDDEN);
-		}
-		elseif ($type == 'error_404')
-		{
-			$response->setStatusCode(Response::HTTP_NOT_FOUND);
-		}
-
-		$response->headers->set('Vary', 'User-Agent', false);
-		$response->headers->set('Content-Type', $content . '; charset=' . \Config::get('characterSet'));
-
-		// Send the cache headers
-		if ($expire !== null && (\Config::get('cacheMode') == 'both' || \Config::get('cacheMode') == 'browser'))
-		{
-			$response->headers->set('Cache-Control', 'private, max-age=' . ($expire - time()));
-			$response->headers->set('Last-Modified', gmdate('D, d M Y H:i:s', time()) . ' GMT');
-			$response->headers->set('Expires', gmdate('D, d M Y H:i:s', $expire) . ' GMT');
-		}
-		else
-		{
-			$response->headers->set('Cache-Control', 'no-store, no-cache, must-revalidate, post-check=0, pre-check=0');
-			$response->headers->set('Last-Modified', gmdate('D, d M Y H:i:s') . ' GMT');
-			$response->headers->set('Expires', 'Fri, 06 Jun 1975 15:10:00 GMT');
-		}
+		@trigger_error('Using Frontend::getResponseFromCache() has been deprecated and will no longer work in Contao 5.0.', E_USER_DEPRECATED);
```

Comment:
I think the `trigger_error` message should also include the information that the method does not work anymore (like the docblock header).